src/main/java/uk/gov/hmcts/befta/util/BeftaUtils.java

```diff
@@ -6,14 +6,18 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 
+import io.restassured.internal.util.IOUtils;
+
 public class BeftaUtils {
 
     public static File getClassPathResourceIntoTemporaryFile(String resourcePath) {
         try {
+            int nameStartsAt = resourcePath.lastIndexOf("/");
+            String simpleName = resourcePath.substring(nameStartsAt + 1);
             InputStream stream = BeftaUtils.class.getClassLoader().getResource(resourcePath).openStream();
-            byte[] buffer = new byte[stream.available()];
-            stream.read(buffer);
-            File tempFile = new File("_temp_" + System.currentTimeMillis() + ".temp");
+            byte[] buffer = IOUtils.toByteArray(stream);
+            File tempFile = new File("_temp_" + System.currentTimeMillis() + "_" + simpleName);
```

Comment:
Are we deleting these temp files after the processing is completed?

Reply:
yes, we are.